tests/pfcwd/test_pfcwd_function.py

```diff
@@ -732,6 +732,73 @@ def storm_detect_path(self, dut, port, action):
+        """
+        Storm detection action and associated verifications
+
+        Args:
+            dut(AnsibleHost) : DUT instance
+            port(string) : DUT port
+            action(string) : PTF test action
+
+        Returns:
+            loganalyzer(Loganalyzer) : instance
+        """
+        restore_time = self.timers['pfc_wd_restore_time_large']
+        detect_time = self.timers['pfc_wd_detect_time']
+
+        loganalyzer = LogAnalyzer(ansible_host=self.dut,
+                                  marker_prefix="pfc_function_storm_detect_{}_port_{}".format(action, port))
+        marker = loganalyzer.init()
+        ignore_file = os.path.join(TEMPLATES_DIR, "ignore_pfc_wd_messages")
+        reg_exp = loganalyzer.parse_regexp_file(src=ignore_file)
+        loganalyzer.ignore_regex.extend(reg_exp)
+        loganalyzer.expect_regex = []
+        loganalyzer.match_regex = []
+
+        if action != "dontcare":
+            start_wd_on_ports(dut, port, restore_time, detect_time, action)
+
+        if not self.pfc_wd['fake_storm']:
+            self.storm_hndle.start_storm()
+
+        if action == "dontcare":
+            start_wd_on_ports(dut, port, restore_time, detect_time, "drop")
+
+        # placing this here to cover all action types. for 'dontcare' action,
+        # wd is started much later after the pfc storm is started
+        if self.pfc_wd['fake_storm']:
+            PfcCmd.set_storm_status(dut, self.queue_oid, "enabled")
+        time.sleep(5)
+
+        # Verify that watchdog is not yet triggered.
+        # test pfcwd functionality on a storm, without traffic.
+        if self.pfc_wd['fake_storm']:
+            loganalyzer.expect_regex.extend([EXPECT_PFC_WD_DETECT_RE + fetch_vendor_specific_diagnosis_re(dut)])
+        else:
+            loganalyzer.match_regex.extend([EXPECT_PFC_WD_DETECT_RE + fetch_vendor_specific_diagnosis_re(dut)])
+        loganalyzer.analyze(marker)
+        marker2 = loganalyzer.init()
+        loganalyzer.expect_regex = []
+        loganalyzer.match_regex = []
+        if self.pfc_wd['fake_storm']:
+            loganalyzer.match_regex.extend([EXPECT_PFC_WD_DETECT_RE + fetch_vendor_specific_diagnosis_re(dut)])
+        else:
+            loganalyzer.expect_regex.extend([EXPECT_PFC_WD_DETECT_RE + fetch_vendor_specific_diagnosis_re(dut)])
+
+        # Watchdog was not triggered until this point.
+        # Finally send traffic and trigger watchdog. Until this point,
+        # there was no data traffic. Only pfc storm was being sent.
+        self.traffic_inst.verify_wd_func(action, self.rx_action, self.tx_action)
+        # storm detect
+        logger.info("Verify if PFC storm is detected on port {}".format(port))
+        loganalyzer.analyze(marker2)
+
+        # test pfcwd functionality on a storm after triggering the watchdog.
+        self.traffic_inst.verify_wd_func(action, self.rx_action, self.tx_action)
+        self.stats.get_pkt_cnts(self.queue_oid, begin=True)
```

Comment:
Why put get_pkt_cnts behind the verify_wd_func?

Reply:
This is the same as older code, I think it is applicable here as well.

Reply:
I don't think so, we need to get the package count first, then call the verify_wd_func to verify the traffic counters.
Please double confirm, thanks

https://github.com/sonic-net/sonic-mgmt/blob/master/tests/pfcwd/test_pfcwd_function.py#L730C1-L733C27
        self.stats.get_pkt_cnts(self.queue_oid, begin=True)
        # test pfcwd functionality on a storm
        self.traffic_inst.verify_wd_func(action, self.rx_action, self.tx_action)

Reply:
@lipxu , let me confirm it. thanks.

Reply:
Any update about it, thanks.

Reply:
And one more question, have you checked the counter value with the background traffic, will it affect the verify_wd_func? thanks

Reply:
@lipxu , In the newest code, I have removed all of these. New code only adds an additional step for cisco-8000 - which is to send some packets using traffic_inst.verify_tx_egress(self.tx_action). This will trigger WD for cisco-8000.
The rest of the code remains same.